src/dbnode/storage/shard.go

```diff
@@ -1409,12 +1409,111 @@ func (s *dbShard) insertSeriesSync(
 	return newEntry, nil
 }
 
+func (s *dbShard) insertNewShardEntriesWithLock(entries []*Entry) {
```

Comment:
add a comment indicating the assumption that entries is sorted.

Reply:
Done